### PR TITLE
Work around for dart-lang/sdk#28599

### PR DESCRIPTION
**Pull request by goderbauer**
Also includes some TODO clean-up.

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 1]
```diff
@@ -217,12 +217,14 @@ class AnsiTerminal {
...
   static const int _ENOTTY = 25;
   static const int _ENETRESET = 102;
+  static const int _ERROR_INVALID_PARAMETER = 87;
 
   /// Setting the line mode can throw for some terminals (with "Operation not
   /// supported on socket"), but the error can be safely ignored.
   static const List<int> _lineModeIgnorableErrors = const <int>[
     _ENOTTY,
     _ENETRESET,
+    _ERROR_INVALID_PARAMETER, // TODO(https://github.com/dart-lang/sdk/issues/28599): remove when bug is fixed
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/base/logger.dart]: Nit: `TODO` format is `TODO(<username>): <comment and/or bug reference>`

**Comment by tvolkert** [comment]
LGTM

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: the bit in brackets should be your username, the URL should go at the end

**Comment by Hixie** [comment]
LGTM.

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 1 above)
Comment (goderbauer) [inline packages/flutter_tools/lib/src/base/logger.dart]: Done.

**Pull request merged**